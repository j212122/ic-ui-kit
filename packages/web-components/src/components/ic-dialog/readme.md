# ic-dialog



<!-- Auto Generated Below -->


## Properties

| Property               | Attribute                 | Description                                                                                                                                                                                 | Type                                                       | Default                                                                                                                                            |
| ---------------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
| `alertHeading`         | `alert-heading`           | If a status is set, sets the heading for the displayed alert.                                                                                                                               | `string`                                                   | `undefined`                                                                                                                                        |
| `alertMessage`         | `alert-message`           | If a status is set, sets the message for the displayed alert.                                                                                                                               | `string`                                                   | `undefined`                                                                                                                                        |
| `buttonProps`          | --                        | Sets the label and onclick functions for default buttons.                                                                                                                                   | `{ label: string; onclick: string; }[]`                    | `[     { label: "Cancel", onclick: "this.hideDialog(); this.icDialogCancel()" },     { label: "Confirm", onclick: "this.icDialogConfirm()" },   ]` |
| `buttons`              | `buttons`                 | If set to `false`, dialog controls will not be displayed overriding buttonProps or slotted dialog controls.                                                                                 | `boolean`                                                  | `true`                                                                                                                                             |
| `closeOnBackdropClick` | `close-on-backdrop-click` | If set to `true`, the dialog will not close when the backdrop is clicked.                                                                                                                   | `boolean`                                                  | `true`                                                                                                                                             |
| `destructive`          | `destructive`             | If default buttons are displayed, sets the 'primary' or rightmost button to the destructive variant. Stops initial focus being set on the 'primary' or rightmost default or slotted button. | `boolean`                                                  | `false`                                                                                                                                            |
| `dismissLabel`         | `dismiss-label`           | Sets the dismiss label tooltip and aria label.                                                                                                                                              | `string`                                                   | `"Dismiss"`                                                                                                                                        |
| `heading` _(required)_ | `heading`                 | Sets the heading for the dialog.                                                                                                                                                            | `string`                                                   | `undefined`                                                                                                                                        |
| `label`                | `label`                   | Sets the optional label for the dialog which appears above the heading.                                                                                                                     | `string`                                                   | `undefined`                                                                                                                                        |
| `size`                 | `size`                    | Sets the maximum and minimum height and width for the dialog.                                                                                                                               | `"large" \| "medium" \| "small"`                           | `"small"`                                                                                                                                          |
| `status`               | `status`                  | If set, displays an alert of the corresponding variant below the heading.                                                                                                                   | `"error" \| "info" \| "neutral" \| "success" \| "warning"` | `undefined`                                                                                                                                        |


## Events

| Event               | Description                                              | Type                |
| ------------------- | -------------------------------------------------------- | ------------------- |
| `icDialogCanceled`  | Cancelation event for default 'Cancel' button behaviour. | `CustomEvent<void>` |
| `icDialogConfirmed` | Confirmation event for default primary button behaviour. | `CustomEvent<void>` |


## Methods

### `hideDialog() => Promise<void>`

Used to dismiss the dialog.

#### Returns

Type: `Promise<void>`



### `showDialog() => Promise<void>`

Used to display the dialog.

#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-button](../ic-button)
- [ic-alert](../ic-alert)

### Graph
```mermaid
graph TD;
  ic-dialog --> ic-typography
  ic-dialog --> ic-button
  ic-dialog --> ic-alert
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  ic-alert --> ic-typography
  ic-alert --> ic-button
  style ic-dialog fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


