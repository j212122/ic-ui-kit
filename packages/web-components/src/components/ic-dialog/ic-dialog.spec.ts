import { newSpecPage } from "@stencil/core/testing";
import { Dialog } from "./ic-dialog";

describe("ic-dialog component", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" buttons="false"></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a label", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" label="Dialog label" buttons="false"></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with slotted content", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" buttons="false">Dialog content</ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render as medium size", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" size="medium" buttons="false"></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render as large size", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" size="large" buttons="false"></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a single default button", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Confirm", onclick: "this.icDialogConfirm()" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should render with two default buttons", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with three default buttons", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Cancel", onclick: "this.hideDialog()" },
      { label: "Options", onclick: "alert('Options...')" },
      { label: "Confirm", onclick: "this.icDialogConfirm()" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should render with no more than three default buttons", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Cancel", onclick: "this.hideDialog()" },
      { label: "Options", onclick: "alert('Options...')" },
      { label: "Confirm", onclick: "this.icDialogConfirm()" },
      { label: "Extra", onclick: "alert('Extra button')" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a destructive default button", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading test" destructive="true"></ic-dialog>',
    });

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Confirm", onclick: "this.icDialogConfirm()" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should render with an alert", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading" status="neutral" alert-heading="Alert heading" alert-message="Alert message" buttons="false"></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with slotted controls", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"><ic-button variant="primary" onclick="alert('Confirmed!')" slot="interaction-controls">Confirm</ic-button></ic-dialog>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should display with the showDialog method", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"></ic-dialog>`,
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;
  });

  it("should hide with the hideDialog method", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"></ic-dialog>`,
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await dialog.hideDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should hide when the escape key is pressed", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"></ic-dialog>`,
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    const event = new KeyboardEvent("keydown", { key: "Escape" });

    await page.rootInstance.handleKeyboard(event);

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should hide when something besides the dialog is clicked", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"></ic-dialog>`,
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    const event = new Event("click");

    await page.rootInstance.handleClick(event);

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should hide when the close button is clicked", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"></ic-dialog>`,
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelector("ic-button").click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should correctly pass onclick functions to a single default button", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [{ label: "Confirm", onclick: "this.hideDialog()" }];

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[1].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should correctly pass onclick functions to two default buttons", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Close", onclick: "this.hideDialog()" },
      { label: "Confirm", onclick: "this.hideDialog()" },
    ];

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[1].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[2].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should correctly pass onclick functions to three default buttons", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [
      { label: "Cancel", onclick: "this.hideDialog()" },
      { label: "Options", onclick: "this.hideDialog()" },
      { label: "Confirm", onclick: "this.hideDialog()" },
    ];

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[1].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[2].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;

    await dialog.showDialog();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeFalsy;

    await page.root.shadowRoot.querySelectorAll("ic-button")[3].click();

    await page.waitForChanges();

    expect(dialog.classList.contains("hidden")).toBeTruthy;
  });

  it("should switch focus with tab", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [{ label: "Confirm", onclick: "this.hideDialog()" }];

    await page.waitForChanges();

    await dialog.showDialog();

    await page.waitForChanges();

    await page.rootInstance.getInteractiveElements();

    await page.rootInstance.setInitialFocus();

    await page.rootInstance.getFocusedElementIndex();

    await page.waitForChanges();

    expect(page.rootInstance.focusedElementIndex).toEqual(0);

    const event = new KeyboardEvent("keydown", { key: "Tab" });

    page.rootInstance.handleKeyboard(event);

    await page.waitForChanges();

    expect(page.rootInstance.focusedElementIndex).toEqual(1);
  });

  it("should switch focus with shift tab", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: '<ic-dialog heading="Dialog heading"></ic-dialog>',
    });

    expect(page.root).not.toBeNull;

    const dialog = document.querySelector("ic-dialog");

    dialog.buttonProps = [{ label: "Confirm", onclick: "this.hideDialog()" }];

    await page.waitForChanges();

    await dialog.showDialog();

    await page.waitForChanges();

    await page.rootInstance.getInteractiveElements();

    await page.rootInstance.setInitialFocus();

    await page.rootInstance.getFocusedElementIndex();

    await page.waitForChanges();

    expect(page.rootInstance.focusedElementIndex).toEqual(0);

    const event = new KeyboardEvent("keydown", { key: "Tab", shiftKey: true });

    page.rootInstance.handleKeyboard(event);

    await page.waitForChanges();

    expect(page.rootInstance.focusedElementIndex).toEqual(1);
  });

  it("should set focus on slotted elements", async () => {
    const page = await newSpecPage({
      components: [Dialog],
      html: `<ic-dialog heading="Dialog heading"><ic-button variant="primary" onclick="alert('Confirmed!')" id="focus" slot="interaction-controls">Confirm</ic-button></ic-dialog>`,
    });

    const dialog = document.querySelector("ic-dialog");

    await dialog.showDialog();

    await page.waitForChanges();

    page.rootInstance.getInteractiveElements();

    page.rootInstance.setInitialFocus();

    page.rootInstance.getFocusedElementIndex();

    expect(page.rootInstance.focusedElementIndex).toBe(0);
  });
});
