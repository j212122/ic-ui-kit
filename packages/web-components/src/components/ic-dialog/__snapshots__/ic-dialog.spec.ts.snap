// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-dialog component should render 1`] = `
<ic-dialog buttons="false" class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render as large size 1`] = `
<ic-dialog buttons="false" class="hidden" heading="Dialog heading" size="large">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog large">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render as medium size 1`] = `
<ic-dialog buttons="false" class="hidden" heading="Dialog heading" size="medium">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog medium">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with a destructive default button 1`] = `
<ic-dialog class="hidden" destructive="true" heading="Dialog heading test">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading test
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" variant="destructive">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with a label 1`] = `
<ic-dialog buttons="false" class="hidden" heading="Dialog heading" label="Dialog label">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label">
                Dialog label
              </ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with a single default button 1`] = `
<ic-dialog class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" data-gets-focus variant="primary">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with an alert 1`] = `
<ic-dialog alert-heading="Alert heading" alert-message="Alert message" buttons="false" class="hidden" heading="Dialog heading" status="neutral">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-alert class="status-alert" heading="Alert heading" id="dialog-alert" message="Alert message" title-above="" variant="neutral"></ic-alert>
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with no more than three default buttons 1`] = `
<ic-dialog class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" variant="secondary">
            Cancel
          </ic-button>
          <ic-button class="dialog-control-button" variant="secondary">
            Options
          </ic-button>
          <ic-button class="dialog-control-button" data-gets-focus variant="primary">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with slotted content 1`] = `
<ic-dialog buttons="false" class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" data-gets-focus variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
    </dialog>
  </mock:shadow-root>
  Dialog content
</ic-dialog>
`;

exports[`ic-dialog component should render with slotted controls 1`] = `
<ic-dialog class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" variant="tertiary">
            Cancel
          </ic-button>
          <ic-button class="dialog-control-button" data-gets-focus variant="primary">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
  <ic-button data-gets-focus onclick="alert('Confirmed!')" slot="interaction-controls" variant="primary">
    Confirm
  </ic-button>
</ic-dialog>
`;

exports[`ic-dialog component should render with three default buttons 1`] = `
<ic-dialog class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls triple-button">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" full-width="" variant="secondary">
            Cancel
          </ic-button>
          <ic-button class="dialog-control-button" full-width="" variant="secondary">
            Options
          </ic-button>
          <ic-button class="dialog-control-button" data-gets-focus full-width="" variant="primary">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;

exports[`ic-dialog component should render with two default buttons 1`] = `
<ic-dialog class="hidden" heading="Dialog heading">
  <mock:shadow-root>
    <dialog aria-describedby="dialog-alert dialog-content" aria-labelledby="dialog-label dialog-heading" class="dialog small">
      <div class="heading-area">
        <div class="heading-content">
          <div class="label">
            <slot name="label">
              <ic-typography id="dialog-label" variant="label"></ic-typography>
            </slot>
          </div>
          <div class="heading">
            <slot name="heading">
              <ic-typography id="dialog-heading" variant="h4">
                Dialog heading
              </ic-typography>
            </slot>
          </div>
        </div>
        <ic-button aria-label="Dismiss" class="close-icon" variant="icon">
          <span class="close-icon-svg" slot="icon">
            svg
          </span>
        </ic-button>
      </div>
      <div class="content-area">
        <ic-typography class="message" id="dialog-content" variant="body">
          <slot></slot>
        </ic-typography>
      </div>
      <div class="dialog-controls">
        <slot name="dialog-controls">
          <ic-button class="dialog-control-button" variant="tertiary">
            Cancel
          </ic-button>
          <ic-button class="dialog-control-button" data-gets-focus variant="primary">
            Confirm
          </ic-button>
        </slot>
      </div>
    </dialog>
  </mock:shadow-root>
</ic-dialog>
`;
